libsplinter/src/admin/store/circuit_proposal.rs

```diff
@@ -74,6 +69,17 @@ impl CircuitProposal {
     pub fn requester_node_id(&self) -> &str {
         &self.requester_node_id
     }
+
+    pub fn builder(&self) -> CircuitProposalBuilder {
```

Comment:
should this be `into_builder` and consume the struct?

Reply:
The builder was added originally because @vaporos  had mentioned that maybe CircuitProposal should be read only. Consuming the struct felt counter to that.